vignettes/relase-process.qmd

```diff
@@ -0,0 +1,198 @@
+---
+title: "The Release Process"
+description: >
+  Reference for how to perform R package releases in the Hubverse
+vignette: >
+  %\VignetteIndexEntry{The Release Process}
+  %\VignetteEngine{quarto::html}
+  %\VignetteEncoding{UTF-8}
+knitr:
+  opts_chunk: 
+    collapse: true
+    comment: '#>'
+---
+
+This document is adapted for The Hubverse from [The Carpentries Hubverse Developer's Handbook](https://carpentries.github.io/workbench-dev/releases.html) (c) The Carpentries. 
+
+## Checklist
+
+ - [ ] Create a new branch from `main` called `release-X.Y.Z`
+ - [ ] Update version number in DESCRIPTION and check that the `Remotes:` is
+       up-to-date
+ - [ ] Update NEWS for the changes in this version
+ - [ ] commit, push changes, and create pull request
```

Comment:
I'm a little confused on what happens if the feature I'm working on is spread across multiple PRs. Do I merge this into `release-*` branch? Also do I increment the dev version? (I would definitely prefer that so we can quickly pin point which version of code we're discussing at any point if needed)

Reply:
> I'm a little confused on what happens if the feature I'm working on is spread across multiple PRs. 
> Do I merge this into `release-*` branch? 

This is a good question. The answer is no. The `release-X.Y.Z` branch is a temporary branch that has one job: to bump the version number. It's created when you want to make a release from `main` and then deleted after it's merged into main and then the GitHub release is created. 

As you work on multiple independent features, you would merge them into `main` and build up to the release. This way, the pull requests can easily indicate how the releases were created. 

See **for example** how the releases are created in The Carpentries: https://github.com/carpentries/sandpaper/releases/tag/0.16.5. In this release, there are seven pull requests that were merged into `main` between April 2024 and June 2024. On 2024-06-18, The `release-0.16.5` branch was created to do the final pull request before release, which revealed minor errors that were easily corrected. Once it was merged, the release was created. 

> Also do I increment the dev version? (I would definitely prefer that so we can quickly pin point which version of code we're discussing at any point if needed)

I've gone back and forth on this. I usually leave the dev version to X.Y.Z.9000 so that we know whether it's dev or release and have people report the output of `packageDescription(pkg)$GitHubSHA1`. With constantly incrementing the version number, there end up being near-constant merge conflicts (and it's often difficult to remember to actually do so).

Reply:
I think this guidance from the R packages book is relevant:

> Increment the development version, e.g. from 9000 to 9001, if you’ve added an important feature and you (or others) need to be able to detect or require the presence of this feature. For example, this can happen when two packages are developing in tandem. This is generally the only reason that we bother to increment the development version. This makes in-development versions special and, in some sense, degenerate. Since we don’t increment the development component with each Git commit, the same package version number is associated with many different states of the package source, in between releases.

Reply:
I've added this guidance to the versioning section